buildpack/build.go

```diff
@@ -81,13 +82,57 @@ func (b *Descriptor) Build(bpPlan Plan, config BuildConfig) (BuildResult, error)
 		return BuildResult{}, err
 	}
 
+	if err := b.processLayerTypes(bpLayersDir, config.Out); err != nil {
+		return BuildResult{}, err
+	}
+
 	if err := b.setupEnv(config.Env, bpLayersDir); err != nil {
 		return BuildResult{}, err
 	}
 
 	return b.readOutputFiles(bpLayersDir, bpPlanPath, bpPlan, config.Out)
 }
 
+func renameLayerDirIfNeeded(layerMetadataFile layertypes.LayerMetadataFile, layerDir string) error {
+	// rename <layers>/<layer> to <layers>/<layer>.ignore if buildpack API >= 0.6 and all of the types flags are set to false
+	if !layerMetadataFile.Launch && !layerMetadataFile.Cache && !layerMetadataFile.Build {
+		if err := os.Rename(layerDir, layerDir+".ignore"); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (b *Descriptor) processLayerTypes(layersDir string, out io.Writer) error {
+	if api.MustParse(b.API).Compare(api.MustParse("0.6")) < 0 {
+		return eachDir(layersDir, b.API, func(path, buildpackAPI string) error {
+			_, msg, err := DecodeLayerMetadataFile(path+".toml", buildpackAPI)
+			if err != nil {
+				return err
+			}
+			if msg != "" {
+				if _, err = out.Write([]byte(msg)); err != nil {
+					return err
+				}
+			}
+			return nil
+		})
+	}
+	return eachDir(layersDir, b.API, func(path, buildpackAPI string) error {
+		layerMetadataFile, msg, err := DecodeLayerMetadataFile(path+".toml", buildpackAPI)
+		if err != nil {
+			return err
+		}
+		if msg != "" {
+			return errors.New(msg)
+		}
```

Comment:
We had this discussion before and I'm not sure.
If anyone has a strong opinion on this, I can make this change, otherwise I'll keep it as is for now.